tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java

```diff
@@ -79,49 +68,35 @@ public static final void derbyNotSupported() {
-     *
-     * @return "MySql"
-     * @deprecated Derby is going away
+     * @Deprecated
      */
-    @Deprecated
-    public static String getDBKind() {
-        return "MySql";
-    }
+    public static final boolean db_Mysql = true;
 
     public String getDBInfo() {
         return dbInfo;
     }
 
-    // === DB workarounds :( - derby by default
-    public static String DB_SQL_IDENTITY = "GENERATED ALWAYS AS IDENTITY";
-    public static String DB_SQL_VARCHARXPATH = "varchar(1024)";
-    public static String DB_SQL_WITHDEFAULT = "WITH DEFAULT";
-    public static String DB_SQL_TIMESTAMP0 = "TIMESTAMP";
-    public static String DB_SQL_CURRENT_TIMESTAMP0 = "CURRENT_TIMESTAMP";
-    public static String DB_SQL_MIDTEXT = "VARCHAR(1024)";
-    public static String DB_SQL_BIGTEXT = "VARCHAR(16384)";
-    public static String DB_SQL_UNICODE = "VARCHAR(16384)"; // unicode type
+    // === DB workarounds
+    public static final String DB_SQL_IDENTITY = "AUTO_INCREMENT PRIMARY KEY";
+    public static final String DB_SQL_VARCHARXPATH = "TEXT(1000)";
+    public static final String DB_SQL_WITHDEFAULT = "DEFAULT";
+    public static final String DB_SQL_TIMESTAMP0 = "DATETIME";
+    public static final String DB_SQL_CURRENT_TIMESTAMP0 = "'1999-12-31 23:59:59'"; // NOW?
+    public static final String DB_SQL_MIDTEXT = "TEXT(1024)";
+    public static final String DB_SQL_BIGTEXT = "TEXT(16384)";
+    public static final String DB_SQL_UNICODE = "BLOB";
     // string
-    public static String DB_SQL_LAST_MOD_TYPE = "TIMESTAMP";
-    public static String DB_SQL_LAST_MOD =
-            " last_mod TIMESTAMP NOT NULL WITH DEFAULT CURRENT_TIMESTAMP  ";
-    public static String DB_SQL_ALLTABLES =
-            "select tablename from SYS.SYSTABLES where tabletype='T'";
-    public static String DB_SQL_BINCOLLATE = "";
-    public static String DB_SQL_ENGINE_INNO = "";
-    public static String DB_SQL_MB4 = "";
+    public static final String DB_SQL_LAST_MOD_TYPE = "TIMESTAMP";
+    public static final String DB_SQL_LAST_MOD =
+            " last_mod TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP ";
+    public static final String DB_SQL_ALLTABLES = "show tables";
+    public static final String DB_SQL_BINCOLLATE = " COLLATE latin1_bin ";
+    public static final String DB_SQL_ENGINE_INNO = "ENGINE=InnoDB";
+    public static final String DB_SQL_MB4 = " CHARACTER SET utf8mb4 COLLATE utf8mb4_bin";
     public static int db_number_open = 0;
     public static int db_number_used = 0;
-    private static int db_UnicodeType = java.sql.Types.VARCHAR; /*
-                                                                * for setNull -
-                                                                * see
-                                                                * java.sql.Types
-                                                                */
+    private static final int db_UnicodeType = java.sql.Types.BLOB;
```

Comment:
On first glance this seems to go beyond dropping derby -- does this potentially change the behavior when using mysql? Now I see setupSqlForServerType had override for mysql... so I'm assuming there's no actual change for mysql

Reply:
Yes it moves the overrides up. I'm trying to make it easier to see what the MySQL settings are.